### PR TITLE
Changed conf property from str to dict in SparkSqlOperator

### LINKED ISSUE #40507: SparkSqlOperator and SparkSubmitOperator are using different types for configurations
### Apache Airflow Provider(s)

apache-spark

### Versions of Apache Airflow Providers

apache-airflow==2.9.2
apache-airflow-providers-apache-spark==4.8.2

### Apache Airflow version

2.9.2

### Operating System

MacOS

### Deployment

Docker-Compose

### Deployment details

_No response_

### What happened

The SparkSubmitOperator uses a dictionary to handle the 'conf' property of the operator
The SparkSqlOperator uses a string in format `PARAM=VALUE,PARAM2=VALUE2` to handle the 'conf' property.

The first option allows a config like this to be passed:
```
conf = {
        'spark.jars.packages': 'org.apache.iceberg:iceberg-spark-runtime-3.5_2.12:1.5.2,org.projectnessie.nessie-integrations:nessie-spark-extensions-3.5_2.12:0.80.0',
        'spark.driver.extraJavaOptions': '-Divy.cache.dir=/tmp -Divy.home=/tmp',
        'spark.sql.extensions': 'org.apache.iceberg.spark.extensions.IcebergSparkSessionExtensions,org.projectnessie.spark.extensions.NessieSparkSessionExtensions'
}
```

while the second option will always split the packages into:
`--conf  spark.jars.packages=org.apache.iceberg:iceberg-spark-runtime-3.5_2.12:1.5.2 --conf org.projectnessie.nessie-integrations:nessie-spark-extensions-3.5_2.12:0.80.0` due to it being split on the comma as a delimiter.
This effectively does not allow adding any of the comma delimited configurations of spark.

The SparkSubmitOperator also has a bigger list of available properties; including the --packages flag which is available as well on the spark/bin/spark-sql script.

### What you think should happen instead

The first option allows for more flexibility when adding configs, and a dictionary seems the right way to store the configs. It would enforce the same behaviour on both spark operators, making it easier to adjust/maintain. Also less documentation to keep :)

https://github.com/apache/airflow/blob/54dfead4ccfe7a5b41eab3433c5a38159434e014/airflow/providers/apache/spark/hooks/spark_sql.py#L146

This is the place where the config is split on ','

### How to reproduce

Create a dag and task: 

```
conf = {
        'spark.jars.packages': 'org.apache.iceberg:iceberg-spark-runtime-3.5_2.12:1.5.2,org.projectnessie.nessie-integrations:nessie-spark-extensions-3.5_2.12:0.80.0',
        'spark.driver.extraJavaOptions': '-Divy.cache.dir=/tmp -Divy.home=/tmp',
        'spark.sql.extensions': 'org.apache.iceberg.spark.extensions.IcebergSparkSessionExtensions,org.projectnessie.spark.extensions.NessieSparkSessionExtensions'
}

config_string = ','.join([f"{key}={value}" for key, value in conf.items()])

merge_branch = SparkSqlOperator(
    name="merge_branch",
    task_id="merge_branch",
    conf=config_string, # requires a string instead of a dict
    conn_id='spark',
    dag=dag,
    sql=f"MERGE BRANCH {ref} INTO main IN nessie",
    retries=0
)

```

### Anything else

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by duhizjame**
Changes the conf property from str to dict in SparkSqlOperator
closes: https://github.com/apache/airflow/issues/40507

The previous PR was closed due to inactivity, I had time this week to fix it

### CODE REVIEW

**Comment by romsharon98** [comment]
@duhizjame the changes looks good.
because it's a breaking change we need to bump the version of the provider (major change) and also add a breaking change caption in the change log.
after this we can merge this change.

**Comment by romsharon98** [comment]
also what do you think about supporting both `str` and `dict` and only parse it differently behind the scene

**Comment by duhizjame** [comment]
@romsharon98 maybe its better that way, so we don't break everyone on versions <= 4.8.2, I will add it to the PR tonight

regarding version changes:
from the contribution docs I cant find who is doing the the version changes; after adding str, we bump only minor version and add a changelog; should I do this or someone from the contributors is doing this after approval?

thanks for the feedback! :)

**Comment by romsharon98** [comment]
> @romsharon98 maybe its better that way, so we don't break everyone on versions <= 4.8.2, I will add it to the PR tonight
> 
> 
> 
> regarding version changes:
> 
> from the contribution docs I cant find who is doing the the version changes; after adding str, we bump only minor version and add a changelog; should I do this or someone from the contributors is doing this after approval?
> 
> 
> 
> thanks for the feedback! :) 

I think that for features we need to change nothing belongs to the provider.yml, I think we have a bot that add the documentation according to the PR title/commit and the version will update by the release manager when we releasing the package. 
I am not sure 100%, probably @eladkal will now better 😃

File: airflow/providers/apache/spark/hooks/spark_sql.py
Comment (romsharon98) [inline airflow/providers/apache/spark/hooks/spark_sql.py]:
```suggestion
                for key, value in conf.items():
                    connection_cmd += ["--conf", f"{key}={value}"]
```

**Comment by duhizjame** [comment]
I have no idea why the pipeline fails to build images, I haven't used breeze to dev/test

**Comment by romsharon98** [comment]
> I have no idea why the pipeline fails to build images, I haven't used breeze to dev/test

You have conflicts that need to be resolved.
After you resolve it we will merge.

**Review by romsharon98 — APPROVED** [review]

**Pull request merged**
